lib/builders/gpu-builder.ts

```diff
@@ -0,0 +1,184 @@
+     * optional, Node IAM Role to be attached to Windows 
+     * and Non-windows nodes.
+     */
+    nodeRole?: iam.Role,
+    /** 
+     * Optional, AMI Type for Windows Nodes
+     */
+    gpuAmiType?: NodegroupAmiType,
+    /** 
+     * Optional, Desired number of nodes to use for the cluster. 
+     */
+    desiredNodeSize?: number,
+    /** 
+     * Optional, Minimum number of nodes to use for the cluster. 
+     */
+    minNodeSize?: number,
+    /** 
+     * Optional, Maximum number of nodes to use for the cluster. 
+     */
+    maxNodeSize?: number,
+    /** 
+     * Optional, Block device size.
+     */
+    blockDeviceSize?: number,
+    /**
+     * Optional, Cluster Provider Tags.
+     */
+    clusterProviderTags?: {
+        [key: string]: string;
+    },
+
+    /**
+     * Optional, Node Group Tags for AL2 nodes 
+     * which run standard cluster software.
+     */
+    nodeGroupTags?: {
+        [key: string]: string;
+    }
+}
+
+/**
+ * Default props to be used when creating the GPU nodes 
+ * for EKS cluster
+ */
+const defaultOptions: GpuOptions = {
+    kubernetesVersion: eks.KubernetesVersion.of("1.27"),
+    instanceClass: ec2.InstanceClass.G5,
+    instanceSize: ec2.InstanceSize.XLARGE,
+    nodeRole: blueprints.getNamedResource("node-role") as iam.Role,
+    gpuAmiType: NodegroupAmiType.AL2_X86_64_GPU,
+    desiredNodeSize: 2,
+    minNodeSize: 2,
+    maxNodeSize: 3,
+    blockDeviceSize: 50,
+    clusterProviderTags: {
+        "Name": "blueprints-gpu-eks-cluster",
+        "Type": "generic-gpu-cluster"
+    },
+    nodeGroupTags: {
+        "Name": "Mng-linux",
+        "Type": "Managed-linux-Node-Group",
+        "LaunchTemplate": "Linux-Launch-Template",
+    }
+  };
+export class GpuBuilder extends blueprints.BlueprintBuilder {
+    /**
+     * This method helps you prepare a blueprint for setting up observability 
+     * returning an array of blueprint addons for AWS managed open source services
+     */
+    public enableGpu(values?: blueprints.Values): GpuBuilder {
+    return this.addOns(
+        new blueprints.addons.AwsLoadBalancerControllerAddOn(),
+        new blueprints.addons.CertManagerAddOn(),
+        new blueprints.addons.CoreDnsAddOn(),
+        new blueprints.addons.KubeProxyAddOn(),
+        new blueprints.addons.VpcCniAddOn(),
+        new blueprints.addons.GpuOperatorAddon({values})
+        );
+    }
+     /**
+     * This method helps you prepare a blueprint for setting up windows nodes with 
+     * usage tracking addon
+     */
+     public static builder(options: GpuOptions): GpuBuilder {
+        const builder = new GpuBuilder();
+        const mergedOptions = merge(defaultOptions, options);
+
+        builder
+            .clusterProvider(
+                new blueprints.GenericClusterProvider({
+                    version: mergedOptions.kubernetesVersion,
+                    tags: mergedOptions.clusterProviderTags,
+                    role: blueprints.getResource(context => {
```

Comment:
Why is this needed? We added this specifically for windows builder to work. If you check the Observability builder, we dont have this.

Reply:
My bad, i was taking argument into builder rather than into the enablegpu function. fixed this